packages/firestore/test/integration/browser/webchannel.test.ts

```diff
@@ -33,41 +45,64 @@ const describeFn =
       xdescribe;
 
 describeFn('WebChannel', () => {
+  it('can auto detect the connection type', done => {
+    const conn = new WebChannelConnection({
+      ...getDefaultDatabaseInfo(),
+      autoDetectLongPolling: true
+    });
+    const stream = conn.openStream<api.ListenRequest, api.ListenResponse>(
+      'Listen',
+      null
+    );
+
+    // Once the stream is open, send an "add_target" request.
+    stream.onOpen(() => stream.send(ADD_TARGET));
+
+    stream.onMessage(() => {});
+
+    // Ensure Time to first byte was raised
+    let timeToFirstByteData: {
+      isLongPollingConnection: boolean;
+      timeToFirstByteMs: number;
+    } | null = null;
+
+    stream.onTimeToFirstByte((isLongPollingConnection, timeToFirstByteMs) => {
+      timeToFirstByteData = { isLongPollingConnection, timeToFirstByteMs };
+      stream.close();
+    });
+
+    stream.onClose(() => {
+      expect(timeToFirstByteData).to.be.ok;
+      expect(timeToFirstByteData!.isLongPollingConnection).to.not.equal(0);
+      expect(timeToFirstByteData!.timeToFirstByteMs).to.be.greaterThan(0);
+      done();
+    });
+  });
+
   it('receives error messages', done => {
-    const projectId = DEFAULT_PROJECT_ID;
-    const info = getDefaultDatabaseInfo();
-    const conn = new WebChannelConnection(info);
+    const conn = new WebChannelConnection(getDefaultDatabaseInfo());
     const stream = conn.openStream<api.ListenRequest, api.ListenResponse>(
       'Listen',
       null
     );
 
     // Test data
     let didSendBadPayload = false;
-    const payload = {
-      database: 'projects/' + projectId + '/databases/(default)',
-      addTarget: {
-        query: {
-          parent: 'projects/' + projectId + '/databases/(default)',
-          structuredQuery: {
-            from: [{ collectionId: 'foo' }]
-          }
-        }
-      }
-    };
 
     // Once the stream is open, send an "add_target" request
     stream.onOpen(() => {
-      stream.send(payload);
+      stream.send(ADD_TARGET);
     });
 
     // Wait until we receive data, then send a bad "addTarget" request, causing
     // the stream to be closed with an error. In this case, bad means having a
     // different database ID.
     stream.onMessage(msg => {
       if (msg.targetChange) {
-        payload.database = 'projects/some-other-project-id/databases/(default)';
         didSendBadPayload = true;
+        const payload = { ...ADD_TARGET };
+        payload.database =
+          ' projects/some-other-project-id/databases/(default)';
```

Comment:
Remove space

Reply:
Sure. FYI, your original code snippet w/ the diff had this in there.

Reply:
It was a test to see if your standards are higher than mine.